### ISSUE TITLE
Warning: nb may be used uninitialized in map.cpp

### ISSUE THREAD

**Issue by mtango688** [minetest/minetest#7033]
##### Issue type

 - Bug report
 
##### Minetest version

```
Minetest 0.5.0-dev-3face01
```

##### OS / Hardware

Operating system:
Linux
CPU:
AMD

GPU model:
OpenGL version:

##### Summary

Compiling gives the following warning:

```
/home/mtango/minetest/src/map.cpp: In member function ‘void Map::transformLiquids(std::map<irr::core::vector3d<short int>, MapBlock*>&, ServerEnvironment*)’:
/home/mtango/minetest/src/map.cpp:520:8: warning: ‘*((void*)& nb +14)’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 struct NodeNeighbor {
        ^
/home/mtango/minetest/src/map.cpp:650:17: note: ‘*((void*)& nb +14)’ was declared here
    NodeNeighbor nb(getNodeNoEx(npos), nt, npos);
                 ^
```

##### Steps to reproduce

git pull
make

**Comment by paramat** [comment]
Probably due to 3face01a202040e4feff3b0936b4aa89c051c98d @juhdanad

**Comment by nerzhul** [comment]
Fixed by 359c8f82ceaef19563d0dba67a090e8fdc2dfb71
